fix: adjust CvFileUploader drop handler to not ignore 'multiple' attribute

felipebritor opened the pull request:
Contributes to #1663 

## What did you do?

Update 'drop' event handler, at CvFileUploader, to handle files considering 'multiple' attribute

## Why did you do it?

Fix the drop behavior when multiple is falsy

## How have you tested it?

Yes. Automated tests also added.

## Were docs updated if needed?

- [ ] N/A
- [X] No
- [ ] Yes

davidnixon commented:
@felipebritor a tweak is still needed. 
1. This does prevent me from selecting multiple files and dropping them on the target. When I do that I only get 1 file as expected. ✅ 
2. I can however drag and drop a file and then drag and drop another file so that now there are two! ❌ 
3. The multiple attribute should default to true ❌ 
4.  The React story is unclear on how the button mode (non drag & drop mode) should work. I wonder if you or @themoenen has an opinion on that?

Referencing the same component in res=act:
https://react.carbondesignsystem.com/?path=/story/components-fileuploader--drag-and-drop-upload-container-example-application&args=multiple:!false

felipebritor commented:
> @felipebritor a tweak is still needed.
> 
> 1. This does prevent me from selecting multiple files and dropping them on the target. When I do that I only get 1 file as expected. ✅
> 2. I can however drag and drop a file and then drag and drop another file so that now there are two! ❌
> 3. The multiple attribute should default to true ❌
> 4. The React story is unclear on how the button mode (non drag & drop mode) should work. I wonder if you or @themoenen has an opinion on that?
> 
> Referencing the same component in res=act: https://react.carbondesignsystem.com/?path=/story/components-fileuploader--drag-and-drop-upload-container-example-application&args=multiple:!false

Regarding your points:
2. Sorry about it, David. Fixed
3. Added to PR. Had to turn multiple into a props for it
4. I think it does [here](https://react.carbondesignsystem.com/?path=/story/components-fileuploader--default). Or did you mean something else?
